virtual_machine/windows/main.tf

```diff
@@ -74,3 +74,81 @@ resource "azurerm_virtual_machine" "virtual_machine" {
     builtby = "Terraform"
   }
 }
+
+resource "azurerm_metric_alertrule" "metric_alertrule_cpu" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-cpu"
+  resource_group_name = "${var.resource_group}"
+  location            = "${var.location}"
+
+  description = "An alert rule to watch the metric Percentage CPU"
+
+  enabled = true
+
+  resource_id = "${element(azurerm_virtual_machine.virtual_machine.*.id, count.index + var.start_index)}"
+  metric_name = "Percentage CPU"
+  operator    = "GreaterThan"
+  threshold   = 90
+  aggregation = "Average"
+  period      = "PT5M"
+
+  email_action {
+    send_to_service_owners = false
+
+    custom_emails = [
+      "${var.support_email}",
+    ]
+  }
+}
+
+resource "azurerm_metric_alertrule" "metric_alertrule_memory" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-memory"
+  resource_group_name = "${var.resource_group}"
+  location            = "${var.location}"
+
+  description = "An alert rule to watch the metric Percentage Memory"
+
+  enabled = true
+
+  resource_id = "${element(azurerm_virtual_machine.virtual_machine.*.id, count.index + var.start_index)}"
+  metric_name = "\\Memory\\Available Bytes"
+  operator    = "LessThan"
+  threshold   = 10
+  aggregation = "Average"
+  period      = "PT5M"
+
+  email_action {
+    send_to_service_owners = false
+
+    custom_emails = [
+      "${var.support_email}",
+    ]
+  }
+}
+
+resource "azurerm_metric_alertrule" "metric_alertrule_diskusage" {
+  count               = "${var.number_servers}"
+  name                = "${var.application_name}_${var.hostname}_vm_${count.index + var.start_index}-diskusage"
+  resource_group_name = "${var.resource_group}"
+  location            = "${var.location}"
+
+  description = "An alert rule to watch the metric Percentage Diskusage"
+
+  enabled = true
+
+  resource_id = "${element(azurerm_virtual_machine.virtual_machine.*.id, count.index + var.start_index)}"
+  metric_name = "\\LogicalDisk(_Total)\\% Free Space"
```

Comment:
Are these backslashes necessary escape characters for something?

Reply:
Yes they are.

Reply:
fixing name standard issues.